File: TechnologiesFundamentals/SoftwareTechnologies/JavaBasics/07.MaxSequenceOfEqualElements/src/MaxSequenceOfEqualElements.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class MaxSequenceOfEqualElements {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        List<Integer> numbers = new ArrayList<>();
        String[] input = scanner.nextLine().split(" ");

        for (String anInput : input) {
            numbers.add(Integer.parseInt(anInput));
        }

        int maxNumbers = 0;
        int count = 1;
        int maxCount = 1;
        int pos = 0;

        while (pos < numbers.size() - 1) {
            if (numbers.get(pos).equals(numbers.get(pos + 1))) {
                count++;

                if (count > maxCount) {
                    maxCount = count;
                    maxNumbers = numbers.get(pos);
                }
            }
            else {
                count = 1;
            }

            pos++;

            if (maxCount == 1) {
                maxNumbers = numbers.get(0);
            }
        }

        for (int i = 0; i < maxCount; i++) {
            System.out.print(maxNumbers + " ");
        }
    }
}

File: TechnologiesFundamentals/ProgrammingFundamentals/List-Excercises/01.MaxSequenceOfEqualElements/01.MaxSequenceOfEqualElements/MaxSequenceOfEqualElements.cs
namespace _01.MaxSequenceOfEqualElements
{
    using System;
    using System.Linq;

    public class MaxSequenceOfEqualElements
    {
        public static void Main(string[] args)
        {
            var numbers = Console.ReadLine()
                .Split(' ')
                .Select(int.Parse)
                .ToList();

            var maxNumbers = 0;
            var count = 1;
            var maxCount = 1;

            for (var i = 0; i < numbers.Count - 1; i++)
            {
                if (numbers[i] == numbers[i + 1])
                {
                    count++;

                    if (count > maxCount)
                    {
                        maxCount = count;
                        maxNumbers = numbers[i];
                    }
                }
                else
                {
                    count = 1;
                }

                if (maxCount == 1)
                {
                    maxNumbers = numbers[0];
                }
            }

            for (var i = 0; i < maxCount; i++)
            {
                Console.Write(maxNumbers);
                Console.Write(" ");
            }
        }
    } 
}